notify/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+            bounded(REGISTER_CHANNEL_SIZE);
+        let (network_alert_sender, network_alert_receiver) = bounded::<Alert>(NOTIFY_CHANNEL_SIZE);
+
+        let mut thread_builder = thread::Builder::new();
+        if let Some(name) = thread_name {
+            thread_builder = thread_builder.name(name.to_string());
+        }
+        let join_handle = thread_builder
+            .spawn(move || loop {
+                select! {
+                    recv(signal_receiver) -> _ => {
+                        break;
+                    }
+                    recv(new_block_register_receiver) -> msg => self.handle_register_new_block(msg),
+                    recv(new_block_receiver) -> msg => self.handle_notify_new_block(msg),
+                    recv(network_alert_register_receiver) -> msg => self.handle_register_network_alert(msg),
+                    recv(network_alert_receiver) -> msg => self.handle_notify_network_alert(msg),
+                }
+            })
+            .expect("Start notify service failed");
+
+        NotifyController {
+            new_block_register,
+            new_block_notifier: new_block_sender,
+            network_alert_register,
+            network_alert_notifier: network_alert_sender,
+            stop: StopHandler::new(SignalSender::Crossbeam(signal_sender), join_handle),
+        }
+    }
+
+    fn handle_register_new_block(
+        &mut self,
+        msg: Result<Request<String, Receiver<BlockView>>, RecvError>,
+    ) {
+        match msg {
+            Ok(Request {
+                responder,
+                arguments: name,
+            }) => {
+                debug!("Register new_block {:?}", name);
+                let (sender, receiver) = bounded::<BlockView>(NOTIFY_CHANNEL_SIZE);
+                self.new_block_subscribers.insert(name, sender);
+                let _ = responder.send(receiver);
+            }
+            _ => debug!("Register new_block channel is closed"),
+        }
+    }
+
+    fn handle_notify_new_block(&mut self, msg: Result<BlockView, RecvError>) {
+        match msg {
+            Ok(block) => {
+                trace!("event new block {:?}", block);
+                // notify all subscribers
+                for subscriber in self.new_block_subscribers.values() {
+                    let _ = subscriber.send(block.clone());
+                }
+                // notify script
+                if let Some(script) = self.config.new_block_notify_script.as_ref() {
+                    let block_json: ckb_jsonrpc_types::BlockView = block.into();
+                    let args =
+                        [serde_json::to_string(&block_json).expect("serializing should be ok")];
+                    if let Err(err) = Command::new(script).args(&args).status() {
+                        error!(
+                            "failed to run new_block_notify_script script, error: {}",
+                            err
+                        );
```

Comment:
It's better to log the `command` and `args` also.

Reply:
Added `command` to the error log, `args` is too large, was skipped. Please review again, thanks.